CADETProcess/optimization/surrogate.py

```diff
@@ -0,0 +1,360 @@
+            ) -> np.ndarray:
+        if return_std and return_cov:
+            raise RuntimeError(
+                "At most one of return_std or return_cov can be requested."
+            )
+
+        surrogate = self.surrogates[eval_fun]['surrogate']
+        x_scaler = self.surrogates[eval_fun]['x_scaler']
+        y_scaler = self.surrogates[eval_fun]['y_scaler']
+
+        X_scaled = x_scaler.transform(X)
+
+        if return_std:
+            _, Y_std_scaled = surrogate.predict(
+                X_scaled, return_std=True
+            )
+            Y_std = Y_std_scaled * y_scaler.scale_
+            return np.array(Y_std, ndmin=2)
+        elif return_cov:
+            _, Y_cov_scaled = surrogate.predict(
+                X_scaled, return_cov=True
+            )
+            Y_cov = Y_cov_scaled * (y_scaler.scale_ ** 2)
+            return np.array(Y_cov, ndmin=2)
+        else:
+            Y_scaled = surrogate.predict(X_scaled)
+            Y = y_scaler.inverse_transform(np.array(Y_scaled, ndmin=2))
+            return Y
+
+    def _update_surrogate_models(self) -> NoReturn:
+        if self.population.n_f > 0:
+            surrogate, x_scaler, y_scaler = self.train_gp(
+                self.population.x, self.population.f
+            )
+            self._update_eval_fun_surrogate(
+                'objectives', surrogate, x_scaler, y_scaler
+            )
+        if self.population.n_g > 0:
+            surrogate, x_scaler, y_scaler = self.train_gp(
+                self.population.x, self.population.g
+            )
+            self._update_eval_fun_surrogate(
+                'nonlinear_constraints', surrogate, x_scaler, y_scaler
+            )
+        if self.population.n_g > 0:
+            surrogate, x_scaler, y_scaler = self.train_gp(
+                self.population.x, self.population.cv
+            )
+            self._update_eval_fun_surrogate(
+                'nonlinear_constraints_violation', surrogate, x_scaler, y_scaler
+            )
+        if self.population.n_m > 0:
+            surrogate, x_scaler, y_scaler = self.train_gp(
+                self.population.x, self.population.m
+            )
+            self._update_eval_fun_surrogate(
+                'meta_scores', surrogate, x_scaler, y_scaler
+            )
+
+    def update(self, population: Population) -> NoReturn:
+        """
+        Update the surrogate model with new population.
+
+        Parameters
+        ----------
+        population : Population
+            New population entries.
+        """
+        self.population.update(population)
+        self._update_surrogate_models()
+
+    def ensures2d(func):
+        """Decorate function to ensure X array is an ndarray with ndmin=2."""
+        @wraps(func)
+        def wrapper(
+                self,
+                X: np.ndarray,
+                *args, **kwargs
+                ) -> Any:
+
+            X = np.array(X)
+            X_2d = np.array(X, ndmin=2)
```

Comment:
Technically, this does not "ensure 2D" (only ndmin=2). Consequently, we should instead reshape to really ensure 2d.

Also, consider moving function out of class since it duplicates what is done in the `OptimizationProblem` class.